sdk/keyvault/azure-security-keyvault-secrets/src/main/java/com/azure/security/keyvault/secrets/implementation/SecretClientImpl.java

```diff
@@ -0,0 +1,934 @@
+     */
+    public Response<KeyVaultSecret> setSecretWithResponse(KeyVaultSecret secret, Context context) {
+        SecretRequestParameters parameters = validateAndCreateSetSecretParameters(secret);
+        context = context == null ? Context.NONE : context;
+        enableSyncRestProxy(context);
+        return service.setSecret(vaultUrl, secret.getName(), apiVersion, ACCEPT_LANGUAGE, parameters,
+                CONTENT_TYPE_HEADER_VALUE, context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
+    }
+
+    private SecretRequestParameters validateAndCreateSetSecretParameters(KeyVaultSecret secret) {
+        Objects.requireNonNull(secret, "The Secret input parameter cannot be null.");
+        SecretRequestParameters parameters = new SecretRequestParameters()
+            .setValue(secret.getValue())
+            .setTags(secret.getProperties().getTags())
+            .setContentType(secret.getProperties().getContentType())
+            .setSecretAttributes(new SecretRequestAttributes(secret.getProperties()));
+        return parameters;
+    }
+
+    /**
+     * Creates a new secret in the key vault asynchronously.
+     * @param name the secret of the secret.
+     * @param value the value of the secret.
+     * @param context the context to use while executing request.
+     * @return A {@link Mono} containing the Response with created secret.
+     */
+    public Mono<Response<KeyVaultSecret>> setSecretWithResponseAsync(String name, String value, Context context) {
+        SecretRequestParameters parameters = new SecretRequestParameters().setValue(value);
+        return service.setSecretAsync(vaultUrl, name, apiVersion, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE))
+            .doOnRequest(ignored -> logger.verbose("Setting secret - {}", name))
+            .doOnSuccess(response -> logger.verbose("Set secret - {}", response.getValue().getName()))
+            .doOnError(error -> logger.warning("Failed to set secret - {}", name, error));
+    }
+
+    /**
+     * Creates a new secret in the key vault.
+     * @param name the secret of the secret.
+     * @param value the value of the secret.
+     * @param context the context to use while executing request.
+     * @return the Response with created secret.
+     */
+    public Response<KeyVaultSecret> setSecretWithResponse(String name, String value, Context context) {
+        SecretRequestParameters parameters = new SecretRequestParameters().setValue(value);
+        enableSyncRestProxy(context);
+        return service.setSecret(vaultUrl, name, apiVersion, ACCEPT_LANGUAGE, parameters, CONTENT_TYPE_HEADER_VALUE,
+                context.addData(AZ_TRACING_NAMESPACE_KEY, KEYVAULT_TRACING_NAMESPACE_VALUE));
+    }
+
+    public Mono<Response<KeyVaultSecret>> getSecretWithResponseAsync(String name, String version, Context context) {
```

Comment:
TODO: Add Javadocs

Reply:
I wouldn't worry too much about this. That javadoc won't be very enlightening anyway.